src/agent/discovery_proxy.cpp

```diff
@@ -251,14 +239,29 @@ void DiscoveryProxy::OnHostDiscovered(const std::string &
 
     hostInfo.mTtl = CapTtl(aHostInfo.mTtl);
 
-    std::for_each(mSubscriptions.begin(), mSubscriptions.end(), [&](const MdnsSubscription &aSubscription) {
-        if (aSubscription.MatchesHost(aHostName))
+    while ((query = otDnssdGetNextQuery(mNcp.GetInstance(), query)) != nullptr)
+    {
+        std::string      hostName, domain;
+        char             queryName[OT_DNS_MAX_NAME_SIZE];
+        otDnssdQueryType type       = otDnssdGetQueryTypeAndName(query, &queryName);
+        otbrError        splitError = OTBR_ERROR_PARSE;
+
+        if (type == OT_DNSSD_QUERY_TYPE_RESOLVE_HOST)
+        {
+            splitError = SplitFullHostName(queryName, hostName, domain);
+        }
+        if (splitError != OTBR_ERROR_NONE)
+        {
+            continue;
+        }
```

Comment:
```suggestion
        otbrError splitError;

        if (type != OT_DNSSD_QUERY_TYPE_RESOLVE_HOST)
        {
            continue;
        }
        
        splitError = SplitFullHostName(queryName, hostName, domain);
        OT_ASSERT(splitError == OT_ERROR_NONE);
```
`SplitFullHostName` should never fail.